asynctools/asyncsync.nim

```diff
@@ -151,7 +151,7 @@ proc fire*(event: AsyncEv) =
   ## Set the internal flag of ``event`` to `true`. All tasks waiting for it
   ## to become `true` are awakened. Task that call `wait()` once the flag is
   ## `true` will not block at all.
-  proc wakeupAll() =
+  proc wakeupAll() {.gcsafe.} =
```

Comment:
Is this 100% gcsafe? If we add this pragma, we need to be 100% sure it is gcsafe otherwise there can be very hard to debug memory corruption issues.

Reply:
It didn't compile without it 🤷 
It was Mamy who told me to add it, but it might not be good, I don't know

I really just want an IPC lib that works haha